demo_usage/convert_tracking_dataset_to_forecasting.py

```diff
@@ -0,0 +1,196 @@
+from argoverse.data_loading.object_label_record import json_label_dict_to_obj_record
+from argoverse.utils.se3 import SE3
+from argoverse.utils.transform import quat2rotmat, quat_argo2scipy
+from recordclass import RecordClass
+from scipy.spatial.distance import cdist
+from scipy.spatial.transform import Rotation
+
+
+class Box3d(RecordClass):
+	center: np.ndarray # x, y, z
+	width: float
+	length: float
+	height: float
+	orientation_q: Tuple[float,float,float,float] # qw, qx, qy, qz
+	name: str
+	track_id: str
+	timestamp: int
+	orientation_yaw: Optional[float] = None
+	v0: Optional[Tuple[float,float,float]] = None # vertex 0
+	v1: Optional[Tuple[float,float,float]] = None # vertex 1
+	v2: Optional[Tuple[float,float,float]] = None # vertex 2
+	v3: Optional[Tuple[float,float,float]] = None # vertex 3
+
+
+def rotmat2quat(R: np.ndarray) -> np.ndarray:
+	"""  """
+	q_scipy =  Rotation.from_matrix(R).as_quat()
+	return quat_scipy2argo(q_scipy)
+
+
+def quat_scipy2argo(q_scipy: np.ndarray) -> np.ndarray:
+	"""Re-order Argoverse's scalar-first [w,x,y,z] quaternion order to Scipy's scalar-last [x,y,z,w]"""
+	x, y, z, w = q_scipy
+	q_argo = np.array([w, x, y, z])
+	return q_argo
+
+
+def quaternion_yaw_scipy(q: Tuple[float,float,float,float]) -> float:
+    """ """
+    q_argo = q
+    q_scipy = quat_argo2scipy(q_argo)
+    yaw, _, _ = Rotation.from_quat(q_scipy).as_euler('zyx')
+    return yaw
+
+
+
+
+def remove_duplicate_instances(instances: List[Box3d]) -> List[Box3d]:
+	"""Remove any duplicate cuboids in ground truth.
+	Any ground truth cuboid of the same object class that shares the same centroid
+	with another is considered a duplicate instance.
+	We first form an (N,N) affinity matrix with entries equal to negative distance.
+	We then find rows in the affinity matrix with more than one zero, and
+	then for each such row, we choose only the first column index with value zero.
+	
+	Args:
+		instances: array of length (M,), each entry is a Box3d
+
+	Returns:
+		array of length (N,) where N <= M, each entry is a unique ObjectLabelRecord
+	"""
+	if len(instances) == 0:
+		return instances
+	assert isinstance(instances, np.ndarray)
+
+	# create affinity matrix as inverse distance to other objects
+	affinity_matrix = compute_affinity_matrix(copy.deepcopy(instances), copy.deepcopy(instances))
+
+	row_idxs, col_idxs = np.where(affinity_matrix == 0)
+	# find the indices where each row index appears for the first time
+	unique_row_idxs, unique_element_idxs = np.unique(row_idxs, return_index=True)
+	# choose the first instance in each column where repeat occurs
+	first_col_idxs = col_idxs[unique_element_idxs]
+	# eliminate redundant column indices
+	unique_ids = np.unique(first_col_idxs)
+	return instances[unique_ids]
+
+
+def compute_affinity_matrix(dts1: List[Box3d], dts2: List[Box3d]) -> np.ndarray:
+	"""Calculate the affinity matrix between detections and ground truth labels,
+	using a specified affinity function type.
```

Comment:
[Nit] Does this function take a configurable affinity function type? If not, we should consider updating this docstring.